R/plotLearnerPredictionPlotly.R

```diff
@@ -0,0 +1,320 @@
+#'   Defaul is \code{NULL}.
+#' @param point.alpha [\code{numeric(1)}]\cr
+#'   For classification: Set the transparancy of prediction point for classification 3D plots with value from 0 to 1.
+#'   Default is 1.
+#' @param err.mark [\code{character(1)}]:
+#'   For classification: Either mark error of the model on the training data (\dQuote{train}) or
+#'   during cross-validation (\dQuote{cv}) or not at all with \dQuote{none}.
+#'   Default is \dQuote{train}.
+#' @param err.size [\code{numeric(1)}]\cr
+#'   For classification: Set misclassified point size.
+#'   Default is \code{pointsize}.
+#' @param err.col [\code{character(1)}]\cr
+#'   For classification: Set the colors of missclassified data points.
+#'   Default value is \code{NULL} with black color.
+#' @param err.alpha [\code{numeric(1)}]\cr
+#'   For classification: Set the transparancy of missclassified data points.
+#'   Default value is \code{point.alpha}.
+#' @template arg_prettynames
+#' @param show.bounding [\code{logical(1)}]\cr
+#'   For classification: Show the bounding region?
+#'   Default is \code{TRUE}.
+#' @param bounding.alpha [\code{numeric(1)}]\cr
+#'   For \code{show.bounding = TRUE}: Set the transparancy of bounding point.
+#'   Default is 0.5.
+#' @return The plotly object.
+#' @importFrom data.table dcast
+#' @importFrom RColorBrewer brewer.pal
+#' @importFrom plotly plot_ly add_trace %>% layout toRGB hide_colorbar hide_legend
+#' @export
+plotLearnerPredictionPlotly = function(learner, task, features = NULL, measures, cv = 10L,  ...,
+                                       gridsize, show.point = TRUE, show.legend = TRUE, show.colorbar = TRUE,
+                                       pointsize = 2, point.col = NULL, point.alpha = 1,
+                                       err.mark = "train", err.size = pointsize, err.col = NULL, err.alpha = point.alpha,
+                                       pretty.names = TRUE, show.bounding = TRUE, bounding.alpha = 0.5) {
+  learner = checkLearner(learner)
+  assert(
+    checkClass(task, "ClassifTask"),
+    checkClass(task, "RegrTask")
+  )
+  td = getTaskDescription(task)
+
+  # features and dimensionality
+  fns = getTaskFeatureNames(task)
+  if (length(fns) == 1L)
+    stopf("plotLearnerPrediction() works with at least 2 features.")
+  if (is.null(features))
+    features = fns[1:2]
+  else {
+    assertCharacter(features, max.len = 3L)
+    assertSubset(features, choices = fns)
+  }
+  taskdim = length(features)
+  if (td$type == "classif" && taskdim %nin% c(2L, 3L))
+    stopf("Classification: currently only 2 or 3 features plots supported in plotLearnerPredictionPlotly(), not: %i", taskdim)
+  if (td$type == "regr" && taskdim != 2L)
+    stopf("Regression: currently only 2 features plots supported in plotLearnerPredictionPlotly(), not: %i", taskdim)
+
+  measures = checkMeasures(measures, task)
+  cv = asCount(cv)
```

Comment:
I would check the argument here rather than converting it and potentially doing something the user doesn't expect.